src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -0,0 +1,291 @@
+        return findCampaignGuide().Book.getPage() == importData["guide"]
+      end,
+      1,
+      function()
+        --log("Campaign Guide import failed!")
+      end
+    )
+  end
+  Wait.time(
+    function() optionPanelApi.loadSettings(importData["options"]) end,
+    0.5
+  )
+  broadcastToAll("Campaign successfully imported!", Color.Green)
+end
+
+
+-- Creates a campaign token with save data encoded into GM Notes based on the current state of the table
+function createCampaignToken(_, playerColor, _)
+  -- clean up chaos tokens
+  blessCurseApi.removeAll(playerColor)
+  chaosBagApi.releaseAllSealedTokens(playerColor)
+
+  local campaignBoxGUID = ""
+  -- find active campaign
+  for _, obj in ipairs(getObjectsWithTag("CampaignBox")) do
+    if obj.type == "Bag" and #obj.getObjects() == 0 then
+      if campaignBoxGUID ~= "" then
+        broadcastToAll("Multiple empty campaign box detected; delete all but one.", Color.Red)
+        return
+      end
+      campaignBoxGUID = obj.getGUID()
+    end
+  end
+  if campaignBoxGUID == "" then
+    broadcastToAll("Campaign box with all placed objects not found!", Color.Red)
+    return
+  end
+  local campaignLog = findCampaignLog()
+  if campaignLog == nil then
+    broadcastToAll("Campaign log not found!", Color.Red)
+    return
+  end
+  local traumaValues = nil
+  local counterData = campaignLog.getVar("ref_buttonData")
+  if counterData ~= nil then
+    traumaValues = {}
+    printToAll("Trauma values found in campaign log!", "Green")
+    for i = 1, 10, 3 do
+      traumaValues[1 + (i - 1) / 3] = counterData.counter[i].value
+      traumaValues[5 + (i - 1) / 3] = counterData.counter[i + 1].value
+    end
+  else
+    printToAll("Trauma values could not be found in campaign log!", "Yellow")
+    printToAll("Default values for health and sanity loaded.", "Yellow")
+  end
+  local campaignGuide = findCampaignGuide()
+  if campaignGuide == nil then
+    broadcastToAll("Campaign guide not found!", Color.Red)
+    return
+  end
+  local campaignGuidePage = campaignGuide.Book.getPage()
+  local campaignData = {
+    box = campaignBoxGUID,
+    log = campaignLog.getData(),
+    bag = chaosBagApi.getChaosBagState(),
+    trauma = traumaValues,
+    decks = deckImporterApi.getUiState(),
+    clueCount = playAreaAPI.getInvestigatorCount(),
+    guide = campaignGuidePage,
+    options = optionPanelApi.getOptions()
+  }
+  campaignTokenData.GMNotes = JSON.encode(campaignData)
+  campaignTokenData.Nickname = os.date("%b %d ") .. getObjectFromGUID(campaignBoxGUID).getName() .. " Save"
+  spawnObjectData({
+    data = campaignTokenData,
+    position = {-21.25, 1.68, 55.59}
+  })
+  broadcastToAll("Campaign successfully exported! Save coin object to import on a fresh save", Color.Green)
+end
+
+
+-- helper functions
+
+function findCampaignLog()
+  if getObjectsWithTag("CampaignLog") then
+    if #getObjectsWithTag("CampaignLog") == 1 then
+    return getObjectsWithTag("CampaignLog")[1]
```

Comment:
Instead of calling `getObjectsWithTag("CampaignLog")` 3x, assign it to a variable and use it for the other two occasions.
Also I don't think you need the outer `if`.

Same principle applies to the campaign guide below. Might actually be worth to turn these into a function that takes the searchtag as parameter to avoid the repetition.